torch/linalg/__init__.py

```diff
@@ -400,6 +400,53 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+solve = _add_docstr(_linalg.linalg_solve, r"""
+linalg.solve(input, other, *, out=None) -> Tensor
+
+Computes the solution ``x`` to the matrix equation ``matmul(input, x) = other``
+with a square matrix, or batches of such matrices, :attr:`input` and multiple right-hand side vectors :attr:`other`.
+If :attr:`input` is batched and :attr:`other` is not, then :attr:`other` is broadcasted
+to have the same batch dimensions as :attr:`input`.
+The resulting tensor has the same shape as broadcasted :attr:`other`.
+
+Supports input of ``float``, ``double``, ``cfloat`` and ``cdouble`` dtypes.
+
+.. note:: If :attr:`input` is a non-invertible matrix or non-square matrix then a RuntimeError will be thrown.
```

Comment:
How does this work with batching? Is it if any matrix in the batch is singular that a runtime error will be thrown?

Reply:
Yes, if at least one matrix is singular then a runtime error is thrown specifying the index of the first singular matrix.

Reply:
https://github.com/pytorch/pytorch/pull/48456/files#diff-7e17421f32124016eb8de04dc2f445da5786a28355e1addc72b305466f590180R1864-R1870

Reply:
Can we add that condition to this sentence like:

"If :attr:\'input\' is a non-square or non-invertible matrix, or a batch containing non-square matrices or one or more non-invertible matrices, then a RuntimeError will be thrown. "

Reply:
Done in da42d0e.